src/general/cpuInfo.go

```diff
@@ -47,30 +47,56 @@ func NewCPULoad() *CPULoad {
 	return &CPULoad{}
 }
 
+// ReadCPULoad reads /proc/stat and returns the average load
+func readCPULoad(c *CPULoad) error {
+	data, error := os.ReadFile("/proc/stat")
+	if error != nil {
+		return error
+	}
+	stringData := string(data)
+	lines := strings.Split(stringData, "\n")
+	// First line should store in the format:
+	// "cpu <usr> <nice> <system> <idle> <iowait> <irq> <softirq> <steal> <guest> <guest_nice>"
+	vals := strings.Split(lines[0], " ")
+	var avg [10]float64
```

Comment:
Maybe use fields instead? Doc [here](https://golang.org/pkg/strings/#Fields).

Edit: This should make iterating over the values simple too as you can slice and start from first index.

Reply:
Done!